File: DSA/Animesh_jha_590016813/Day_17_question_3.py
def integer_square_root(n):
    if n < 2:
        return n  

    left = 1
    right = n
    result = 1

    while left <= right:
        mid = left + (right - left) // 2

        if mid * mid == n:
            return mid
        elif mid * mid < n:
            result = mid 
            left = mid + 1
        else:
            right = mid - 1

    return result

n = 49
print(integer_square_root(n))

File: DSA/VaniJain_590012020/Day17Question3.java
import java.util.*;
public class IntegerSquareRoot 
{
    public static void main(String[] args) 
    {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter a positive number: ");
        int n = scanner.nextInt();
        if (n < 0) 
        {
            System.out.println("Please enter a non-negative number.");
            return;
        }
        int left = 0, right = n;
        int result = -1;
        while (left <= right) 
        {
            int mid = left + (right - left) / 2;
            long square = (long) mid * mid;
            if (square == n) 
            {
                result = mid;
                break;
            } else if (square < n) 
            {
                result = mid; 
                left = mid + 1;
            } else {
                right = mid - 1;
            }
        }
        System.out.println("Integer square root of " + n + " is: " + result);
    }
}
